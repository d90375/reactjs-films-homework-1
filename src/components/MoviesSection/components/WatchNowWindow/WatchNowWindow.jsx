import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Button from '../../../Button';

import styles from './WatchNowWindow.scss';

class WatchNowWindow extends Component {
  handleClick = async () => {
    const { film: { id }, fetchMovie } = this.props;
    await fetchMovie(id);
  };

  openViewInfo = async () => {

  }

  render() {
    return (
      <div className={styles.container}>
        <button className={styles.watch} type="button" onClick={this.handleClick}>&#9658;</button>
        <p>Watch Now</p>
        <Button onClick={this.openViewInfo} color="secondary">View Info</Button>
      </div>
    );
  }
}

WatchNowWindow.propTypes = {
  film: PropTypes.shape({
    id: PropTypes.number,
  }).isRequired,
  fetchMovie: PropTypes.func.isRequired,
};

export default WatchNowWindow;
