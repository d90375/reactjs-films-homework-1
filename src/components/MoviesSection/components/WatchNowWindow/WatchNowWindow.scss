@import '../../../../mixins.scss';

.container {
  position: absolute;
  top: 0;
  display: none;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 250px;
  height: 220px;
  margin: 0 10px;
  font-size: 16px;
  color: white;
  border-radius: 6px 6px 0 0;
  background-color: rgba($color: #000, $alpha: 0.7);
  box-sizing: border-box;
}

.watch {
  padding: 0 14px 0 18px;
  font-size: 26px;
  line-height: 60px;
  background-color: #3bb4e0;
  color: white;
  border: none;
  border-radius: 50%;
  cursor: pointer;

  @include outline-none;

  &:hover {
    background-color: #3bb4e0cb;
  }
}