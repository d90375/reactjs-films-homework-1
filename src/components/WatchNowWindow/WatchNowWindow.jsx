import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';

import styles from './WatchNowWindow.scss';

const WatchNowWindow = ({ switchViewInfo, film: { id }, fetchTrailer }) => (
  <div className={styles.container}>
    <button className={styles.watch} type="button" onClick={() => fetchTrailer(id)}>&#9658;</button>
    <p>Watch Now</p>
    <Button onClick={switchViewInfo} color="secondary">View Info</Button>
  </div>
);

WatchNowWindow.propTypes = {
  film: PropTypes.shape({
    id: PropTypes.number,
  }).isRequired,
  fetchTrailer: PropTypes.func.isRequired,
  switchViewInfo: PropTypes.func.isRequired,
};

export default WatchNowWindow;
