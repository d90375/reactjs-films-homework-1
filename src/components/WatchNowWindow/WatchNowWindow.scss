@import '../../mixins.scss';

.container {
  position: absolute;
  top: 0;
  display: none;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 250px;
  height: 220px;
  font-size: 16px;
  color: white;
  border-radius: 6px 6px 0 0;
  background-color: rgba($color: #000, $alpha: 0.7);
  box-sizing: border-box;

  &:hover {
    display: flex;
  }

  @media (min-width: 390px) and (max-width: 659px) {
    width: 100%;
    height: calc(100% - 100px);
  }
}

.watch {
  padding: 0 14px 0 18px;
  font-size: 26px;
  line-height: 60px;
  background-color: #3bb4e0;
  color: white;
  border: none;
  border-radius: 50%;
  cursor: pointer;

  @include outline-none;

  &:hover {
    background-color: #3bb4e0cb;
  }
}
