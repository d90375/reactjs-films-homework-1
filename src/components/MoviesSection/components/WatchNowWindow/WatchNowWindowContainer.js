import { connect } from 'react-redux';
import { fetchTrailer } from '../../../../modules/trailer/trailerAction';
import WatchNowWindow from './WatchNowWindow';

const mapDispatchToProps = {
  fetchTrailer,
};

export default connect(
  null,
  mapDispatchToProps,
)(WatchNowWindow);
