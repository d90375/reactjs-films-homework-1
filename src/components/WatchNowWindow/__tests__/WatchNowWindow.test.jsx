import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { create } from 'react-test-renderer';
import WatchNowWindow from '../WatchNowWindow';

describe('WatchNowWindow tests', () => {
  const film = {
    id: 123,
  };

  describe('WatchNowWindow render', () => {
    it('WatchNowWindow renders correctly', () => {
      const mockCallBack = jest.fn();
      const renderer = new ShallowRenderer();
      renderer.render(<WatchNowWindow
        switchViewInfo={mockCallBack}
        fetchTrailer={mockCallBack}
        film={film}
      />);
      const result = renderer.getRenderOutput();
      expect(result).toMatchSnapshot();
    });
  });

  describe('onClick function', () => {
    it('onClick callback function called', () => {
      const mockCallBack = jest.fn();

      const tree = create(
        <WatchNowWindow
          switchViewInfo={mockCallBack}
          fetchTrailer={mockCallBack}
          film={film}
        >
          Trending
        </WatchNowWindow>,
      );

      const button = tree.root.findAllByProps({ type: 'button' })[0];
      button.props.onClick();

      expect(mockCallBack.mock.calls.length).toEqual(1);
      expect(mockCallBack).toHaveBeenCalledWith(123);
    });
  });
});
