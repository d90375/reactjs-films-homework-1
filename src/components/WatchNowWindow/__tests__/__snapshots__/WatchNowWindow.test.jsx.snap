// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatchNowWindow tests WatchNowWindow render WatchNowWindow renders correctly 1`] = `
<div
  className="container"
>
  <button
    className="watch"
    onClick={[Function]}
    type="button"
  >
    â–º
  </button>
  <p>
    Watch Now
  </p>
  <Button
    color="secondary"
    onClick={[MockFunction]}
  >
    View Info
  </Button>
</div>
`;
